### PR TITLE
Handle lense_dir based on $::rubysitedir.

### PR DESCRIPTION
**Pull request by bzed**
Fixes: #65

### CODE REVIEW

**Comment by mcanevet** [comment]
@bzed could you please fix unit tests? Tests are passing on master so there must be something weird in your PR that make them fail.

**Comment by bzed** [comment]
@mcanevet I gave it a short try, but I fail to understand why it fails. So please take it and fix it or leave it broken.

File: spec/classes/augeas_spec.rb [hunk 1]
```diff
@@ -14,7 +14,11 @@
     end
   end
 
-  lens_dir = Puppet.version < '4.0.0' ? '/usr/share/augeas/lenses' : '/opt/puppetlabs/puppet/share/augeas/lenses'
+  if Puppet.version >= '4.0.0' and facts[:rubysitedir] =~ /\/opt\/puppetlabs\/puppet/
```
Comment (mcanevet) [inline spec/classes/augeas_spec.rb]: This test is misplaced, the `facts` hash is not available here

File: spec/defines/augeas_lens_spec.rb [hunk 2]
```diff
@@ -3,7 +3,11 @@
 describe 'augeas::lens' do
   let (:title) { 'foo' }
 
-  lens_dir = Puppet.version < '4.0.0' ? '/usr/share/augeas/lenses' : '/opt/puppetlabs/puppet/share/augeas/lenses'
+  if Puppet.version >= '4.0.0' and facts[:rubysitedir] =~ /\/opt\/puppetlabs\/puppet/
```
Comment (mcanevet) [inline spec/defines/augeas_lens_spec.rb]: Same here

**Comment by mcanevet** [comment]
@bzed the unit tests failure definitely comes from your code. See my comments.

**Comment by bzed** [comment]
@mcanevet see my last comment. take it, fix it, or leave it broken. I don't have the time to bother with that mess call spec tests.

**Comment by bzed** [comment]
@mcanevet so whats your fix for #65 ?

**Comment by mcanevet** [comment]
@bzed I think your code would work.
I just closed your PR because it breaks the unit tests (which is unacceptable) and you obviously don't want to fix that (even if it's trivial).
If you don't have time "to bother with that mess call spec tests", then I can provide you the contact of our salesman. I'm sure he will be happy to provide you a quote so that we do it for you. :-)

**Comment by bzed** [comment]
@mcanevet well thanks, I can happily live with the broken spec tests - and with a fork. but I guess other users might be happy if they could just keep using your module.

**Comment by mcanevet** [comment]
Stop trolling and fix your damn tests idiot!

**Comment by raphink** [comment]
@bzed having tests that pass is a common requirement for all serious open-source projects. In the current case, it really shouldn't be too hard to fix, especially as everything is already implemented and the tests pass on the master branch.

We understand other users might want to use this patch (although we don't) but we are not willing to maintain code that is untested, and even less code that breaks code that we need (even if it fixes another issue). If you need support on how to fix the tests, I'm sure you can find it on many IRC/slack Puppet channels, and we might even help you fix them if you ask, but we won't merge a PR that doesn't pass existing tests.

**Pull request closed without merging**